test/e2e/actuals_test.go

```diff
@@ -253,6 +253,7 @@ func workResourceIdentifiers() []placementv1beta1.ResourceIdentifier {
 func crpStatusUpdatedActual(
 	wantSelectedResourceIdentifiers []placementv1beta1.ResourceIdentifier,
 	wantSelectedClusters, wantUnselectedClusters []string,
+	wantObservedResourceIndex string,
```

Comment:
```suggestion
	wantObservedResourceIndex int,
```

can you use the int as the input type? Using string type as the index is kind of weird.

Reply:
The ObservedResourceIndex field in the API is string https://github.com/Azure/fleet/blob/main/apis/placement/v1beta1/clusterresourceplacement_types.go#L353 that's the reason I went with string.

If we pass int we have to convert to string though